cli/src/diff_util.rs

```diff
@@ -495,7 +496,20 @@ pub fn show_color_words_diff(
     async {
         while let Some((path, diff)) = diff_stream.next().await {
             let ui_path = path_converter.format_file_path(&path);
-            let (left_value, right_value) = diff?;
+            let (left_value, right_value) = match diff {
+                Err(jj_lib::backend::BackendError::ReadAccessDenied {
+                    object_type: _,
+                    hash: _,
+                    source,
+                }) => {
+                    writeln!(
+                        formatter.labeled("access-denied"),
+                        "Access denied to {ui_path}: {source}"
+                    )?;
+                    continue;
+                }
+                other => other?,
```

Comment:
nit: It might be better to handle `ReadAccessDenied` at upper layer (so the downstream impls are forced to deal with `Ok(MaterializedTreeValue::AccessDenied)` for example), but I don't know. That probably depends on how exceptional the error is.

Mercurial appears to have a config knob to turn censored data to `b""`, but I don't know if that's a good idea.